src/Microsoft.ML.Transforms/CategoricalHashTransform.cs

```diff
@@ -135,100 +136,123 @@ public sealed class Arguments : TransformInputBase
         /// <param name="hashBits">Number of bits to hash into. Must be between 1 and 30, inclusive.</param>
         /// <param name="invertHash">Limit the number of keys used to generate the slot name to this many. 0 means no invert hashing, -1 means no limit.</param>
         /// <param name="outputKind">The type of output expected.</param>
-        public static IDataTransform Create(IHostEnvironment env,
+        public static IDataView Create(IHostEnvironment env,
             IDataView input,
             string name,
             string source = null,
-            int hashBits = Defaults.HashBits,
-            int invertHash = Defaults.InvertHash,
-            CategoricalTransform.OutputKind outputKind = Defaults.OutputKind)
+            int hashBits = CategoricalHashEstimator.Defaults.HashBits,
+            int invertHash = CategoricalHashEstimator.Defaults.InvertHash,
+            CategoricalTransform.OutputKind outputKind = CategoricalHashEstimator.Defaults.OutputKind)
         {
-            var args = new Arguments()
-            {
-                Column = new[] { new Column(){
-                    Source = source ?? name,
-                    Name = name
-                    }
-                },
-                HashBits = hashBits,
-                InvertHash = invertHash,
-                OutputKind = outputKind
-            };
-            return Create(env, args, input);
+            return new CategoricalHashEstimator(env, name, source, outputKind).Fit(input).Transform(input) as IDataView;
         }
 
         public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
```

Comment:
>public [](start = 8, length = 6)

can we make this private?

Reply:
We call it in entrypoint, so internal

---
In reply to: [220398909](https://github.com/dotnet/machinelearning/pull/1033#discussion_r220398909) [](ancestors = 220398909)